server/container_create_linux.go

```diff
@@ -585,6 +571,14 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 			}
 		}
 	}()
+
+	// TODO: eventually, this should be in the container package, but it's going through a lot of churn
+	// and SpecAddAnnotations is already passed too many arguments
+	ctr.Config().Annotations, err = s.Runtime().FilterDisallowedAnnotations(sb.RuntimeHandler(), ctr.Config().Annotations)
```

Comment:
Ok, this is probably not a big deal, but this invocation seems to suggest `FilterDisallowedAnnotations` is NOT mutating its argument - because we assign the return value. But it seems `FilterDisallowedAnnotations` is actually mutating the argument? Do we want to make this behaviour more explicit? perhaps either returning a copy or returning only error, so it's very clear the function mutates its argument.